test/utils/browser.js

```diff
@@ -30,32 +52,83 @@ exports.openMusic = function (options) {
   // Execute many async steps
   before(function startBrowser () {
     this.browser = wd.remote();
+    // Uniquely identify a browserstack build based on environment variable
+    if (process.env.USE_BROWSERSTACK) {
+      this.browser = wd.remote('hub.browserstack.com', 80);
+    }
     global.browser = this.browser;
   });
   before(function openBrowser (done) {
-    this.browser.init({browserName: 'chrome'}, done);
+    var that = this;
+    // DEV: Force the Windows 8 platform as it appears to run our test more
+    //      stable than other platforms
+    this.browser.init({
+      browserName: 'chrome',
+      name: 'google-music.js Selenium tests',
+      project: 'google-music.js Selenium tests',
+      platform: 'WIN8',
+      'browserstack.user': BROWSERSTACK_USER,
+      'browserstack.key': BROWSERSTACK_KEY
+    }, function setBrowserVariables () {
+      // DEV: Maximize the browser window to minimize the possiblity of "element not visible erros"
+      that.browser.maximize();
+      that.browser.setAsyncScriptTimeout(ASYNC_SCRIPT_TIMEOUT, done);
+    });
   });
   before(function navigateToMusicBeforeLogin (done) {
     this.browser.get(url, done);
   });
-  before(function handleLoginViaCookies (done) {
+  before(function handleLogin (done) {
     var browser = this.browser;
-    async.forEach(cookies, function setCookies (cookie, cb) {
-      // If the cookie is not for .google.com, skip it
-      // DEV: As discovered by Burp suite's repeater, we only need `SID`, `HSID`, `SSID` but this is simpler
-      if (cookie.domain !== '.google.com') {
-        process.nextTick(cb);
-      // Otherwise, set it
-      } else {
-        browser.setCookie(cookie, cb);
+    var that = this;
+
+    async.waterfall([
+      function findSignInButton (cb) {
+        browser.waitForElementByCssSelector('[data-action=signin]', asserters.isDisplayed, cb);
+      },
+      function clickSignInButton (el, cb) {
+        el.click(cb);
+      },
+      function findEmailInput (cb) {
+        browser.waitForElementById('Email', asserters.isDisplayed, cb);
+      },
+      function enterEmailIntoInput (el, cb) {
+        exports.simpleExecute.call(that, function setEmailInputValue () {
+          document.getElementById('Email').value = arguments[0];
+        }, [GOOGLE_MUSIC_JS_EMAIL], cb);
+      },
+      function findNextButton (success, cb) {
```

Comment:
After writing about `simpleExecute`, I'm realizing that this is mostly a misnamed parameter. Please rename it to `result` or `executeResult`. No need to assert against it but we need to remove confusion.